src/components/FormConfigurator.tsx

```diff
@@ -1,27 +1,22 @@
 import { SwapMode } from '@jup-ag/react-hook';
 import classNames from 'classnames';
-import React from 'react';
+import React, { useEffect, useRef } from 'react';
 import { FormState, UseFormReset, UseFormSetValue } from 'react-hook-form';
 import ChevronDownIcon from 'src/icons/ChevronDownIcon';
 import InfoIconSVG from 'src/icons/InfoIconSVG';
-import { IFormConfigurator } from 'src/pages/_app';
-import { DEFAULT_EXPLORER, FormProps, IInit } from 'src/types';
 import Toggle from './Toggle';
 import Tooltip from './Tooltip';
 import { AVAILABLE_EXPLORER } from '../contexts/preferredExplorer/index';
+import { IFormConfigurator, INITIAL_FORM_CONFIG } from 'src/constants';
+import { useRouter } from 'next/router';
 
-const templateOptions: { name: string; description: string; values: FormProps }[] = [
+const templateOptions: { name: string; description: string; values: IFormConfigurator }[] = [
   {
     name: 'Default',
     description: 'Full functionality and swap experience of Terminal.',
     values: {
-      swapMode: SwapMode.ExactIn,
-      initialAmount: '',
-      fixedAmount: false,
-      initialInputMint: '',
-      fixedInputMint: false,
-      initialOutputMint: '',
-      fixedOutputMint: false,
+      ...INITIAL_FORM_CONFIG,
+      formProps: INITIAL_FORM_CONFIG.formProps,
```

Comment:
this might copy reference and might mutate the initial form config

Reply:
it's all primitives now, but added a freeze to INITIAL_FORM_CONFIG, would it help?

Reply:
yea freeze would help

Reply:
but it only works for outer object, not nested inside. i think best is to just clone the obj

```
      formProps: {...INITIAL_FORM_CONFIG.formProps},
```